src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookings.java

```diff
@@ -111,28 +112,38 @@ public EventBooking add(final String eventId, final Long userId, final BookingSt
     }
 
     @Override
-    public void updateStatus(final String eventId, final Long userId, final BookingStatus status, final Map<String, String> additionalEventInformation) throws SegueDatabaseException {
+    public void updateStatus(final String eventId, final Long userId, final Long reservedById, final BookingStatus status, final Map<String, String> additionalEventInformation) throws SegueDatabaseException {
         PreparedStatement pst;
 
         try (Connection conn = ds.getDatabaseConnection()) {
 
             if (additionalEventInformation != null) {
                 pst = conn.prepareStatement("UPDATE event_bookings " +
-                    "SET status = ?, updated = ?, additional_booking_information = ?::text::jsonb " +
+                    "SET status = ?, updated = ?, reserved_by = ?, additional_booking_information = ?::text::jsonb " +
                     "WHERE event_id = ? AND user_id = ?;");
                 pst.setString(1, status.name());
                 pst.setTimestamp(2, new java.sql.Timestamp(new Date().getTime()));
-                pst.setString(3, objectMapper.writeValueAsString(additionalEventInformation));
-                pst.setString(4, eventId);
-                pst.setLong(5, userId);
+                try {
+                    pst.setLong(3, reservedById);
+                } catch (NullPointerException e) {
+                    pst.setNull(3, Types.INTEGER);
+                }
+                pst.setString(4, objectMapper.writeValueAsString(additionalEventInformation));
+                pst.setString(5, eventId);
+                pst.setLong(6, userId);
             } else {
                 pst = conn.prepareStatement("UPDATE event_bookings " +
-                    "SET status = ?, updated = ? " +
+                    "SET status = ?, updated = ?, reserved_by = ? " +
                     "WHERE event_id = ? AND user_id = ?;");
                 pst.setString(1, status.name());
                 pst.setTimestamp(2, new java.sql.Timestamp(new Date().getTime()));
-                pst.setString(3, eventId);
-                pst.setLong(4, userId);
+                try {
+                    pst.setLong(3, reservedById);
```

Comment:
Surely this could be an `if`, rather than a try-catch? Java is not Python, where exceptions are preferred control flow :P

Reply:
#6aede53c